source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1139 +25,3368 @@
 
-#ifdef dnsMAX_REQUEST_ATTEMPTS
-    #error now called: ipconfigDNS_REQUEST_ATTEMPTS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigUDP_TASK_PRIORITY
-    #error now called: ipconfigIP_TASK_PRIORITY
-#endif
+/*
+ * Used to define away static keyword for CBMC proofs
+ */
 
-#ifdef ipconfigUDP_TASK_STACK_SIZE_WORDS
-    #error now called: ipconfigIP_TASK_STACK_SIZE_WORDS
+#ifndef _static
+    #define _static    static
 #endif
 
-#ifdef ipconfigDRIVER_INCLUDED_RX_IP_FILTERING
-    #error now called: ipconfigETHERNET_DRIVER_FILTERS_PACKETS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigMAX_SEND_BLOCK_TIME_TICKS
-    #error now called: ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS
-#endif
+/*
+ * Since all code is made compatible with the MISRA rules, the inline functions
+ * disappear. Normally defined in portmacro.h or FreeRTOSConfig.h.
+ */
 
-#ifdef ipconfigUSE_RECEIVE_CONNECT_CALLBACKS
-    #error now called: ipconfigUSE_CALLBACKS
+#ifndef portINLINE
+    #define portINLINE    inline
 #endif
 
-#ifdef ipconfigNUM_NETWORK_BUFFERS
-    #error now called: ipconfigNUM_NETWORK_BUFFER_DESCRIPTORS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigTCP_HANG_PROT
-    #error now called: ipconfigTCP_HANG_PROTECTION
-#endif
+/*
+ * Used to standardize macro checks since ( MACRO == 1 ) and ( MACRO != 0 )
+ * are used inconsistently.
+ */
 
-#ifdef ipconfigTCP_HANG_PROT_TIME
-    #error now called: ipconfigTCP_HANG_PROTECTION_TIME
+#ifndef ipconfigENABLE
+    #define ipconfigENABLE    ( 1 )
 #endif
 
-#ifdef FreeRTOS_lprintf
-    #error now called: FreeRTOS_debug_printf
+#ifndef ipconfigDISABLE
+    #define ipconfigDISABLE    ( 0 )
 #endif
 
-#ifdef  ipconfigBUFFER_ALLOC_FIXED_SIZE
-    #error ipconfigBUFFER_ALLOC_FIXED_SIZE was dropped and replaced by a const value, declared in BufferAllocation[12].c
+#ifndef ipconfigIS_ENABLED
+    #define ipconfigIS_ENABLED( x )    ( x != ipconfigDISABLE )
 #endif
 
-#ifdef  ipconfigNIC_SEND_PASSES_DMA
-    #error now called: ipconfigZERO_COPY_TX_DRIVER
+#ifndef ipconfigIS_DISABLED
+    #define ipconfigIS_DISABLED( x )    ( x == ipconfigDISABLE )
 #endif
 
-#ifdef  HAS_TX_CRC_OFFLOADING
-    #error now called: ipconfigHAS_TX_CRC_OFFLOADING
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef  HAS_RX_CRC_OFFLOADING
-    #error now called: ipconfigHAS_RX_CRC_OFFLOADING
-#endif
+/*
+ * pdFREERTOS_ERRNO_EAFNOSUPPORT
+ *
+ * Address family not supported by protocol.
+ *
+ * Note: Now included in FreeRTOS-Kernel/projdefs.h, so this serves as a
+ * temporary kernel version check. To be removed in a future version.
+ */
 
-#ifdef ipconfigTCP_RX_BUF_LEN
-    #error ipconfigTCP_RX_BUF_LEN is now called ipconfigTCP_RX_BUFFER_LENGTH
-#endif
+/*#ifndef pdFREERTOS_ERRNO_EAFNOSUPPORT
```

Comment:
Incorrect Indentation

Reply:
Aside from that one macro missing in older kernel versions, is there actually a particular minimum required kernel version? I didn't want to enforce an arbitrary minimum

Reply:
Can we remove it?